File: src/main/java/com/realworld/project/user/dto/RequestRegisterUser.java
package com.realworld.project.user.dto;

import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.Getter;

import javax.validation.constraints.NotEmpty;

@Getter
@JsonRootName("user")
public class RequestRegisterUser {
    @NotEmpty
    private String username;

    @NotEmpty
    private String email;

    @NotEmpty
    private String password;
}

File: kotlin/src/main/kotlin/com/example/realworld/application/user/api/dto/RequestRegisterUser.kt
package com.example.realworld.application.user.api.dto

import com.fasterxml.jackson.annotation.JsonRootName
import javax.validation.constraints.NotEmpty

@JsonRootName("user")
data class RequestRegisterUser(
    @field:NotEmpty
    val username: String,

    @field:NotEmpty
    val email: String,

    //for test
    @field:NotEmpty
    val password: String?= null
)
